sv_pipeline/genome/load_data.py

```diff
@@ -0,0 +1,176 @@
+    other_fields = {
+        'xstart': rows.xpos,
+        'xstop': hl.if_else(hl.is_defined(rows.info.END2),
+                            chrom_offset.get(rows.info.CHR2.replace('^chr', '')) + rows.info.END2, rows.xpos),
+        'sortedTranscriptConsequences': rows._sortedTranscriptConsequences.flatmap(
+            lambda x: x.genes.map(lambda y: hl.struct(gene_symbol=y, gene_id=gene_id_mapping[y],
+                                                      predicted_consequence=x.predicted_consequence))),
+        'transcriptConsequenceTerms': [rows.svType],
+        'svTypeDetail': hl.if_else(rows.svType == 'CPX', rows.info.CPX_TYPE, rows._svDetail_ins),
+        'cpxIntervals': hl.if_else(hl.is_defined(rows.info.CPX_INTERVALS),
+                                   rows.info.CPX_INTERVALS.map(lambda x: hl.struct(type=x.split('_chr')[0],
+                                                              chrom=x.split('_chr')[1].split(':')[0],
+                                                              start=hl.int32(x.split(':')[1].split('-')[0]),
+                                                              end=hl.int32(x.split('-')[1]))),
+                                   hl.null(interval_type)),
+    }
+
+    # remove empty temp fields
+    other_fields.update({field[1:]: hl.if_else(hl.len(rows[field]) > 0, rows[field],
+                                               hl.null(hl.dtype('array<str>'))) for field in temp_fields})
+    rows = rows.annotate(**other_fields)
+
+    rows = rows.rename({'rsid': 'variantId'})
+
+    fields = basic_fields + list(other_fields.keys()) + ['genotypes']
+    return rows.key_by('variantId').select(*fields)
+
+
+def main():
+    p = argparse.ArgumentParser()
+    p.add_argument('input_dataset', help='input VCF file')
+    p.add_argument('--matrixtable-path', default='', help='path for Hail MatrixTable data')
+    p.add_argument('--skip-sample-subset', action='store_true')
+    p.add_argument('--ignore-missing-samples', action='store_true')
+    p.add_argument('--project-guid')
+    p.add_argument('--gencode-release', default=29)
+    p.add_argument('--sample-type', default='WGS')
+    p.add_argument('--es-host', default='localhost')
+    p.add_argument('--es-port', default='9200')
+    p.add_argument('--num-shards', default=6)
+    p.add_argument('--block-size', default=2000)
+
+    args = p.parse_args()
+
+    if args.matrixtable_path == '':
+        args.matrixtable_path = '{}.mt'.format(args.input_dataset.split('.vcf')[0])
+
+    hl.init()
+
+    global gene_id_mapping
+    gene_id_mapping = hl.literal(load_gencode(args.gencode_release, genome_version=args.sample_type))
+
+    # For the CMG dataset, we need to do hl.import_vcf() for once for all projects.
+    if os.path.isdir(args.matrixtable_path):
+        logger.info('Use the existing MatrixTable {}.'.format(args.matrixtable_path))
+    else:
+        hl.import_vcf(args.input_dataset, reference_genome='GRCh38').write(args.matrixtable_path)
```

Comment:
The matrix table is better to be written for once because the MT file can be reused for different projects. 
If a new VCF file with the same file name is used, the existing matrix table must be deleted. Maybe, I should add a reminder message here to remind deleting the MT file when needed.
Any intermediate writing or querying on the annotation result will cause performance loss. The cost of opening an MT file is very low because it just opens a file and does nothing else.

Reply:
currently we want this to behave the way the exome SV pipeline behaves. This means running it locally against a local file and NOT on dataproc against a google bucket file. For this first pass at the piepline, running it in the cloud is out of scope. If in the future we feel like we need to move this to the cloud, we should update file storage then. But in the mean time it is not a good use of your time to work on this optimization

Reply:
also worth noting that this whole pipeline, including export, runs in less than 5 minutes. So we are on a totally different scale than the SNP pipeline in terms of optimization/ resiliency